WordPress/src/test/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModelTest.kt

```diff
@@ -1,51 +1,131 @@
 package org.wordpress.android.ui.activitylog.list.filter
 
+import com.nhaarman.mockitokotlin2.anyOrNull
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyNoMoreInteractions
+import com.nhaarman.mockitokotlin2.whenever
 import kotlinx.coroutines.InternalCoroutinesApi
 import org.assertj.core.api.Assertions.assertThat
 import org.junit.Before
 import org.junit.Test
+import org.mockito.Mock
 import org.wordpress.android.BaseUnitTest
 import org.wordpress.android.TEST_DISPATCHER
 import org.wordpress.android.test
 import org.wordpress.android.ui.activitylog.list.filter.ActivityLogTypeFilterViewModel.ListItemUiState
 import org.wordpress.android.ui.activitylog.list.filter.ActivityLogTypeFilterViewModel.UiState
-import org.wordpress.android.ui.activitylog.list.filter.ActivityLogTypeFilterViewModel.UiState.Content
+import org.wordpress.android.ui.activitylog.list.filter.DummyActivityTypesProvider.DummyActivityType
+import org.wordpress.android.ui.activitylog.list.filter.DummyActivityTypesProvider.DummyAvailableActivityTypesResponse
 
 @InternalCoroutinesApi
 class ActivityLogTypeFilterViewModelTest : BaseUnitTest() {
     private lateinit var viewModel: ActivityLogTypeFilterViewModel
+    @Mock private lateinit var dummyActivityTypesProvider: DummyActivityTypesProvider
 
     @Before
     fun setUp() {
-        viewModel = ActivityLogTypeFilterViewModel(TEST_DISPATCHER, TEST_DISPATCHER)
+        viewModel = ActivityLogTypeFilterViewModel(dummyActivityTypesProvider, TEST_DISPATCHER, TEST_DISPATCHER)
     }
 
     @Test
     fun `fullscreen loading shown, when screen initialized`() = test {
-        val uiStates = initObservers().uiStates
+        val uiStates = init().uiStates
 
-        viewModel.start()
+        viewModel.start(0L)
 
         assertThat(uiStates[0]).isInstanceOf(UiState.FullscreenLoading::class.java)
     }
 
     @Test
-    fun `section header gets added as first item in the list`() {
-        val uiStates = initObservers().uiStates
+    fun `available activity types fetched, when screen initialized`() = test {
+        init()
 
-        viewModel.start()
+        viewModel.start(0L)
 
-        assertThat((viewModel.uiState.value as Content).items[0])
+        verify(dummyActivityTypesProvider).fetchAvailableActivityTypes(anyOrNull())
+        verifyNoMoreInteractions(dummyActivityTypesProvider)
```

Comment:
Question (❓ ): How much is `verifyNoMoreInteractions(...)` in general is utilized in our tests and when?

Reply:
Not much - I add it when I want to test that a certain action results in certain re-action. Eg. when I invoke two methods in a test and I want to make sure that only one of them results in an API request. However, it doesn't make sense in this test - I think I kept updating the test and stopped making sense after. 

Removed in c2e74b17d3b14ab62f601cd5e3c8e926dad58a64

Reply:
Great, great, thanks for the explanation. Yeap, I agree, I used to do this as well but lately I find myself doing less and less this kind of verification and only applying them when indeed it makes total sense and there is no ambiguity of its value. But, my general rule of thumb is that those are usually not needed much.